service/parserprovider/properties.go

```diff
@@ -26,28 +25,30 @@ import (
 	"go.opentelemetry.io/collector/config"
 )
 
-type setFlagProvider struct{}
+type propertiesProvider struct {
+	props []string
+}
 
-// NewSetFlag returns a ParserProvider, that provides a config.Map from set flag(s) properties.
+// NewPropertiesProvider returns a ParserProvider, that provides a config.Map from set flag(s) properties.
 //
-// The implementation reads set flag(s) from the cmd and concatenates them as a "properties" file.
-// Then the properties file is read and properties are set to the loaded Parser.
-func NewSetFlag() ParserProvider {
-	return &setFlagProvider{}
+// The implementation parses the properties and concatenates them as a "properties" file.
+// Then the properties file is read and properties are set to the returned config.Map.
+func NewPropertiesProvider(props []string) ParserProvider {
```

Comment:
Adding a comment here about the `props` argument format would help (something like "key-value list separated by equal sign with a `.` as key delimiter" maybe?)

Reply:
Done.